specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/virtualWan.json

```diff
@@ -2634,6 +2634,231 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnGateways/{gatewayName}/vpnConnections/{connectionName}/vpnLinkConnections/{linkConnectionName}/sharedKey": {
+      "get": {
+        "operationId": "VpnLinkConnections_SharedKey_Get",
+        "description": "Get the shared key of VpnLink connection specified.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "gatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the gateway."
+          },
+          {
+            "name": "connectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn connection."
+          },
+          {
+            "name": "linkConnectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn link connection."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. Returns the shared key of VpnLink connection.",
+            "schema": {
+              "$ref": "#/definitions/ConnectionSharedKey"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "VpnSiteLinkConnectionSharedKeyGet": {
+            "$ref": "./examples/VpnSiteLinkConnectionSharedKeyGet.json"
+          }
+        }
+      },
+      "post": {
```

Comment:
You can define it as PUT for creating "/sharedKeys/{sharedKeyName}" resource. Why are using Post API here?

Reply:
We are following the API spec for Virtual Network Gateway Set Shared Key for consistency. Please refer to the below link

https://learn.microsoft.com/en-us/rest/api/network-gateway/virtual-network-gateway-connections/set-shared-key?view=rest-network-gateway-2023-09-01&tabs=HTTP

Example: PUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/connections/{virtualNetworkGatewayConnectionName}/sharedkey?api-version=2023-09-01

It uses the PUT and "/connections/{virtualNetworkGatewayConnectionName}/sharedkey/", but we got the comment from API spec reviewers that Set/Reset Key operation should both be implemented as POST action. So, we changed it to POST action. Please refer to this API spec document for more information.
https://microsoft-my.sharepoint.com/:w:/p/liusheng/ESV5acTQcilMhbdQM3ZiCnIBBdbfEnZwUK4XfxhIzXgWTg?e=dK3wa3&nav=eyJjIjo4NDIzNDQyMjl9

Reply:
This needs to be fixed. You can't put POST on a resource type. If you want to implement this as a POST action, it needs to be a verb and applied to a resource instance, not a [broken] resource type. You could use this path for this POST action:

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnGateways/{gatewayName}/vpnConnections/{connectionName}/vpnLinkConnections/{linkConnectionName}/resetSharedKey

Reply:
Sheng already provided response. We will definitely follow this in future but can we get an exception for this time ?